pkg/operator/staticpodoperator/controlloops/installer_controller.go

```diff
@@ -0,0 +1,377 @@
+		return nil, err
+	}
+
+	ret := currKubeletState.DeepCopy()
+	ret.TargetDeploymentID = deploymentID
+	ret.Errors = nil
+
+	return ret, nil
+}
+
+func getInstallerPodImage() string {
+	if val := os.Getenv("OPERATOR_IMAGE"); len(val) > 0 {
+		return val
+	}
+
+	return "docker.io/openshift/origin-cluster-kube-apiserver-operator:v4.0"
+}
+
+func (c InstallerController) sync() error {
+	operatorSpec, originalOperatorStatus, resourceVersion, err := c.operatorConfigClient.Get()
+	if err != nil {
+		return err
+	}
+	operatorStatus := originalOperatorStatus.DeepCopy()
+
+	switch operatorSpec.ManagementState {
+	case operatorv1alpha1.Unmanaged:
+		return nil
+	case operatorv1alpha1.Removed:
+		// TODO probably just fail.  Static pod managers can't be removed.
+		return nil
+	}
+
+	requeue, syncErr := c.createInstallerController(operatorSpec, operatorStatus, resourceVersion)
+	if requeue && syncErr == nil {
+		return fmt.Errorf("synthetic requeue request")
+	}
+	err = syncErr
+
+	if err != nil {
+		if !reflect.DeepEqual(originalOperatorStatus, operatorStatus) {
+			v1alpha1helpers.SetOperatorCondition(&operatorStatus.Conditions, operatorv1alpha1.OperatorCondition{
+				Type:    operatorv1alpha1.OperatorStatusTypeFailing,
+				Status:  operatorv1alpha1.ConditionTrue,
+				Reason:  "StatusUpdateError",
+				Message: err.Error(),
+			})
+			if _, updateError := c.operatorConfigClient.UpdateStatus(resourceVersion, operatorStatus); updateError != nil {
+				glog.Error(updateError)
+			}
+		}
+		return err
+	}
+
+	return nil
+}
+
+// Run starts the kube-apiserver and blocks until stopCh is closed.
+func (c *InstallerController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	glog.Infof("Starting InstallerController")
+	defer glog.Infof("Shutting down InstallerController")
+
+	// doesn't matter what workers say, only start one.
+	go wait.Until(c.runWorker, time.Second, stopCh)
+
+	<-stopCh
+}
+
+func (c *InstallerController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *InstallerController) processNextWorkItem() bool {
+	dsKey, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(dsKey)
+
+	err := c.sync()
+	if err == nil {
+		c.queue.Forget(dsKey)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", dsKey, err))
+	c.queue.AddRateLimited(dsKey)
+
+	return true
+}
+
+// eventHandler queues the operator to check spec and status
+func (c *InstallerController) eventHandler() cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.queue.Add(installerControllerWorkQueueKey) },
+		UpdateFunc: func(old, new interface{}) { c.queue.Add(installerControllerWorkQueueKey) },
+		DeleteFunc: func(obj interface{}) { c.queue.Add(installerControllerWorkQueueKey) },
+	}
+}
+
+const installerPod = `apiVersion: v1
+kind: Pod
+metadata:
+  namespace: openshift-kube-apiserver
```

Comment:
this need to be parametrized

Reply:
can we turn this into a go template? (use assets to manage it maybe?)

Reply:
> can we turn this into a go template? (use assets to manage it maybe?)

Assets maybe.  Can I get it to be a potential followup?